### DISCUSSION TITLE
Taxonomy/Terminology

### DISCUSSION THREAD

**Discussion by buckynbrocko** [tree-sitter/tree-sitter#3516] [General]
tldr: I want to contribute a top-down explanation of tree-sitter/grammars to the documentation.

I'm confused or unsure of my understanding for some things - I'm sure whatever clarification I need could be useful in the documentation at large. I'll periodically be adding more responses with my current understanding to this discussion thread, but this entry is mostly to get  my thoughts into words.

Things I've been confused about:
- grammar projects and how different steps of generated artifacts relate to each other
  - `src/` vs `build/` vs `bindings/`
  -  `grammar.js` vs `grammar.json` vs `node-types.json`
  - `tree-sitter generate` vs `tree-sitter generate --build`
- canonical packaging of a parser
  - what constitutes a parser
  - how are parsers meant to be distributed/consumed

**Comment by clason** [comment]
Do you have actual questions? That might be a more productive approach here.

**Reply by buckynbrocko** [reply]
The end-goal is a holistic description of the parts and processes involved in using tree-sitter, but yes, I intend to expand on the points individually into more concrete questions. The first discussion entry is more of a stub.

**Comment by sogaiu** [comment]
FWIW, there is some info that is related mentioned among the questions (and related "discussion") [here](https://github.com/sogaiu/ts-questions).

Also of interest might be some of the links mentioned in [this issue](https://github.com/tree-sitter/tree-sitter/issues/1309).
